File: app/src/main/java/com/example/influencer/Features/OnBoarding___CountryAndDateSelector/UI/OnBoarding/OnboardingViewModel.java
package com.example.influencer.Features.OnBoarding___CountryAndDateSelector.UI.OnBoarding;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.influencer.Core.Utils.Event;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;

@HiltViewModel
public class OnboardingViewModel extends ViewModel {

    private final MutableLiveData<Event<Boolean>> _navigateToHome = new MutableLiveData<>();
    public LiveData<Event<Boolean>> navigateToHome  = _navigateToHome ;

    @Inject
    public OnboardingViewModel() {
        //no dependencies needed
    }

    public void StartnowSelected(){
        _navigateToHome.postValue(new Event<>(true));
    }
}

File: app/src/main/java/com/example/influencer/Features/OnBoarding___CountryAndDateSelector/UI/OnBoarding/OnboardingViewModel.kt
package com.example.influencer.Features.OnBoarding___CountryAndDateSelector.UI.OnBoarding

import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject
import androidx.lifecycle.ViewModel
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.LiveData
import com.example.influencer.Core.Utils.Event

@HiltViewModel
class OnboardingViewModel @Inject constructor() : ViewModel() {

    private val _navigateToHome = MutableLiveData<Event<Unit>>()
    val navigateToHome: LiveData<Event<Unit>> = _navigateToHome

    fun startNowSelected() {
        _navigateToHome.value = Event(Unit)
    }
}